### ISSUE TITLE
hi , iam not able send rquest with cookie in another dart file

### ISSUE THREAD

**Issue by aslamibnabdulla** [flutterchina/dio#1357]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by jgoyvaerts** [comment]
Please provide an error message and a complete code sample of what is going wrong

**Comment by aslamibnabdulla** [comment]
import 'package:dio/dio.dart';
import 'package:dio_cookie_manager/dio_cookie_manager.dart';
import 'package:cookie_jar/cookie_jar.dart';
import 'package:flutter/cupertino.dart';

class NetworkHelper {
  var cookieJar = CookieJar();
  var dio = Dio();

  Future logIn(String username, String password) async {
    dio.interceptors.add(CookieManager(cookieJar));
    var response = await dio
        .post(url, data: {
      "username": username.toString(),
      "password": password.toString()
    });
   
    return response.data;
  }

  Future getData() async {
   
    var data = await dio
        .get(url);
    
    return data.data;
  }

  
}
   

and here is my onpressed in the login page



 onPressed: () async {
                          setState(() {
                            isLoading = true;
                          });
                          print(email);
                          var response = await networkHelper.logIn(
                              email.toString(), password.toString());
                          print(response.toString());

                          await Navigator.push(
                            context,
                            MaterialPageRoute(
                              builder: ((context) => HomePage(
                                    emailID: email.toString(),
                                    password: password,
                                  )),
                            ),
                          );
                          setState(() {
                            isLoading = false;
                          });

                          
                        },








here is my  initstate from the nextscreen where i want to display the requested data


void initState() {
    // TODO: implement initState
    networkHelper.getData();

    super.initState();
  }





here is the errror i am getting          E/flutter (25238): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: DioError [DioErrorType.response]: Http status error [401]

**Comment by kuhnroyal** [comment]
Not a clear and reproducible sample.
Make sure you use the same instance of `networkHelper` and maybe add the interceptor when initializing Dio.

```dart
var dio = Dio()..interceptors.add(CookieManager(cookieJar));
```

**Comment by aslamibnabdulla** [comment]
> Not a clear and reproducible sample. Make sure you use the same instance of `networkHelper` and maybe add the interceptor when initializing Dio.
> 
> ```dart
> var dio = Dio()..interceptors.add(CookieManager(cookieJar));
> ```

**Comment by aslamibnabdulla** [comment]
Thanks for the help
